packager/rbi_gen.cc

```diff
@@ -1321,4 +1322,58 @@ void RBIGenerator::run(core::GlobalState &gs, const UnorderedSet<core::ClassOrMo
         });
     threadBarrier.Wait();
 }
+
+void RBIGenerator::runSinglePackage(core::GlobalState &gs,
+                                    const UnorderedSet<core::ClassOrModuleRef> &packageNamespaces,
+                                    core::NameRef package, string outputDir, WorkerPool &workers) {
+    auto output = runOnce(gs, package, packageNamespaces);
+    if (!output.rbi.empty()) {
+        FileOps::write(absl::StrCat(outputDir, "/", output.baseFilePath, ".package.rbi"), output.rbi);
+        FileOps::write(absl::StrCat(outputDir, "/", output.baseFilePath, ".deps.json"), output.rbiPackageDependencies);
+    }
+
+    if (!output.testRBI.empty()) {
+        FileOps::write(absl::StrCat(outputDir, "/", output.baseFilePath, ".test.package.rbi"), output.testRBI);
+        FileOps::write(absl::StrCat(outputDir, "/", output.baseFilePath, ".test.deps.json"),
+                       output.testRBIPackageDependencies);
+    }
+}
+
+RBIGenerator::SinglePackageInfo RBIGenerator::findSinglePackage(core::GlobalState &gs, std::string packageName) {
+    SinglePackageInfo res;
+
+    auto &db = gs.packageDB();
+    auto &packages = db.packages();
+
+    auto nameParts = absl::StrSplit(packageName, "::");
+    auto nameSize = std::distance(nameParts.begin(), nameParts.end());
+    for (auto pkg : packages) {
+        auto &info = db.getPackageInfo(pkg);
+
+        auto &fullName = info.fullName();
+        if (nameSize < fullName.size()) {
+            continue;
+        }
+
+        bool equal = true;
+        auto it = fullName.begin();
+        for (auto part : nameParts) {
+            if (it == fullName.end() || it->shortName(gs) != part) {
+                equal = false;
+                break;
+            }
+
+            ++it;
+        }
```

Comment:
I feel like there ought to be a `std::` function for this, but I'm not seeing it.

Reply:
I wanted to use `std::equal`, but not being able to supply a function for the equality check breaks it. Yet another case where importing range-v3 would help :)

Reply:
It turns out that `std::equal` does take a binary predicate as of c++17, so I reworked this to use that instead.